### PR TITLE
Fegundos will now explode when moving against diagonal gentryii chain…

### PR DESCRIPTION
**Pull request by EvidentlyCube**
…s + tests

[Relevant thread](http://forum.caravelgames.com/viewtopic.php?TopicID=40957)

### CODE REVIEW

File: DRODLib/DbRooms.cpp [hunk 1]
```diff
@@ -2826,8 +2826,8 @@ const
 }
 
 //*****************************************************************************
-bool CDbRoom::DoesSquarePreventDiagonal(
-//Does a 4-connected space prevent diagonal movement?
+bool CDbRoom::DoesOrthoSquarePreventDiagonal(
+//Does an ortho-square prevent diagonal movement between the two two tiles?
```
Comment (mrimer) [inline DRODLib/DbRooms.cpp:2830]: "two two" --> "two"

File: DRODLib/Fegundo.cpp [hunk 2]
```diff
@@ -191,3 +190,25 @@ void CFegundo::Explode(CCueEvents &CueEvents)
 
 	room.ConvertUnstableTar(CueEvents);
 }
+
+
+//*****************************************************************************
+bool CFegundo::DoesMovePreventExplosion(
+// Checks if obstacles during a given move can prevent explosions from occuring
+	const UINT wX, const UINT wY,  //(in) Coords of square from which to move.  Must be valid.
+	const int dx, const int dy  //(in) Directin coords to move
```
Comment (mrimer) [inline DRODLib/Fegundo.cpp:199]: "directin" -> "direction"

File: DRODLib/Fegundo.cpp [hunk 3]
```diff
@@ -191,3 +190,25 @@ void CFegundo::Explode(CCueEvents &CueEvents)
...
+)
+{
+
+	// Arrows prevent explosion even if they are placed on top of an obstacle, even if they are
+	// behind gentryii chain diagonal
+	if (DoesArrowPreventMovement(wX, wY, dx, dy))
+		return true;
+
+	// Ditto for orto squares, must NOT use DoesSquarePreventDiagonal, otherwise gentryii chain
```
Comment (mrimer) [inline DRODLib/Fegundo.cpp:208]: "orto" -> "ortho"

File: DRODLib/Fegundo.cpp [hunk 4]
```diff
@@ -191,3 +190,25 @@ void CFegundo::Explode(CCueEvents &CueEvents)
...
+
+
+//*****************************************************************************
+bool CFegundo::DoesMovePreventExplosion(
+// Checks if obstacles during a given move can prevent explosions from occuring
+	const UINT wX, const UINT wY,  //(in) Coords of square from which to move.  Must be valid.
+	const int dx, const int dy  //(in) Directin coords to move
+)
+{
+
```
Comment (mrimer) [inline DRODLib/Fegundo.cpp:202]: Remove empty line

**Review by mrimer — APPROVED** [review]
Thank you for the fix!
Posted some comment clean-up.

**Comment by EvidentlyCube** [comment]
@mrimer did the changes

**Comment by mrimer** [comment]
Thank you!  Says there's a project file conflict in the test project.  Would you please resolve?

**Comment by EvidentlyCube** [comment]
Resolved now!

**Review by mrimer — APPROVED** [review]

**Pull request merged**